### PR TITLE
Format alert message

### PR DESCRIPTION
**Pull request by anmolbabu**
tendrl-bug-id: Tendrl/alerting#32
Signed-off-by: anmolbabu <anmolbudugutta@gmail.com>

### CODE REVIEW

**Comment by anmolbabu** [comment]
Needs to be verified...

**Comment by anmolbabu** [comment]
Verified

**Comment by anmolbabu** [comment]
@r0h4n @nthomas-redhat @shtripat Please review

**Comment by anmolbabu** [comment]
@r0h4n @nthomas-redhat @shtripat Please review

File: tendrl/alerting/handlers/cluster/brick_utilization_handler.py [hunk 1]
```diff
@@ -10,6 +10,48 @@ class BrickUtilizationHandler(AlertHandler):
     def __init__(self):
         AlertHandler.__init__(self)
 
+    def format_alert_message(self):
+        if (
+            'message' not in self.alert.tags and
```
Comment (shtripat) [inline tendrl/alerting/handlers/cluster/brick_utilization_handler.py]: no check required for plugin_instance here?

File: tendrl/alerting/handlers/cluster/cluster_utilization_handler.py [hunk 2]
```diff
@@ -10,6 +10,35 @@ class ClusterUtilizationHandler(AlertHandler):
     def __init__(self):
         AlertHandler.__init__(self)
 
+    def format_alert_message(self):
+        if (
+            'message' not in self.alert.tags and
```
Comment (shtripat) [inline tendrl/alerting/handlers/cluster/cluster_utilization_handler.py]: no check required for plugin_instance here?

File: tendrl/alerting/handlers/cluster/osd_utilization_handler.py [hunk 3]
```diff
@@ -10,6 +10,37 @@ class OSDUtilizationHandler(AlertHandler):
     def __init__(self):
         AlertHandler.__init__(self)
 
+    def format_alert_message(self):
+        if (
+            'message' not in self.alert.tags and
```
Comment (shtripat) [inline tendrl/alerting/handlers/cluster/osd_utilization_handler.py]: no check required for plugin_instance here?

File: tendrl/alerting/handlers/cluster/pool_utilization_handler.py [hunk 4]
```diff
@@ -10,6 +10,37 @@ class PoolUtilizationHandler(AlertHandler):
     def __init__(self):
         AlertHandler.__init__(self)
 
+    def format_alert_message(self):
+        if (
+            'message' not in self.alert.tags and
```
Comment (shtripat) [inline tendrl/alerting/handlers/cluster/pool_utilization_handler.py]: no check required for plugin_instance here?

File: tendrl/alerting/handlers/cluster/volume_utilization_handler.py [hunk 5]
```diff
@@ -10,6 +10,37 @@ class VolumeUtilizationHandler(AlertHandler):
     def __init__(self):
         AlertHandler.__init__(self)
 
+    def format_alert_message(self):
+        if (
+            'message' not in self.alert.tags and
```
Comment (shtripat) [inline tendrl/alerting/handlers/cluster/volume_utilization_handler.py]: no check required for plugin_instance here?

File: tendrl/alerting/handlers/node/cpu_handler.py [hunk 6]
```diff
@@ -10,6 +10,35 @@ class CpuHandler(AlertHandler):
     def __init__(self):
         AlertHandler.__init__(self)
 
+    def format_alert_message(self):
+        if (
+            'message' not in self.alert.tags and
```
Comment (shtripat) [inline tendrl/alerting/handlers/node/cpu_handler.py]: no check required for plugin_instance here?

File: tendrl/alerting/handlers/node/memory_handler.py [hunk 7]
```diff
@@ -10,6 +10,35 @@ class MemoryHandler(AlertHandler):
     def __init__(self):
         AlertHandler.__init__(self)
 
+    def format_alert_message(self):
+        if (
+            'message' not in self.alert.tags and
```
Comment (shtripat) [inline tendrl/alerting/handlers/node/memory_handler.py]: no check required for plugin_instance here?

File: tendrl/alerting/handlers/node/mount_point_handler.py [hunk 8]
```diff
@@ -10,6 +10,37 @@ class MountPointHandler(AlertHandler):
     def __init__(self):
         AlertHandler.__init__(self)
 
+    def format_alert_message(self):
+        if (
+            'message' not in self.alert.tags and
```
Comment (shtripat) [inline tendrl/alerting/handlers/node/mount_point_handler.py]: no check required for plugin_instance here?

File: tendrl/alerting/handlers/node/swap_handler.py [hunk 9]
```diff
@@ -10,6 +10,35 @@ class SwapHandler(AlertHandler):
     def __init__(self):
         AlertHandler.__init__(self)
 
+    def format_alert_message(self):
+        if (
+            'message' not in self.alert.tags and
```
Comment (shtripat) [inline tendrl/alerting/handlers/node/swap_handler.py]: no check required for plugin_instance here?

File: tendrl/alerting/handlers/cluster/brick_utilization_handler.py [hunk 10]
```diff
@@ -10,6 +10,48 @@ class BrickUtilizationHandler(AlertHandler):
     def __init__(self):
         AlertHandler.__init__(self)
 
+    def format_alert_message(self):
```
Comment (shtripat) [inline tendrl/alerting/handlers/cluster/brick_utilization_handler.py]: Implementation of this function looks more or less similar in all the cases. Only thing changing is name/description of the plugin instance. Can we try to make it generic and have implemented in base class itself with messages appending plugin instance name/description?

Comment (anmolbabu) [inline tendrl/alerting/handlers/cluster/brick_utilization_handler.py]: @shtripat no the way plugin_instance is used is varying and that is the major factor influencing the message formation..

The reason this is happening is:
1. Stats from collectd read plugins(the stats collection plugins in our case) are passed to collectd write plugins(the write_graphite plugin in our case) by using collectd#values#dispatch
now the collectd#Values does not allow us to pass a lot of meta data which makes it difficult to later figure out the hierarchy of a particular resource for which a utilization alert is detected by collectd. The only where such info fits in collecd scheme of things is collectd#Values#plugin_instance.
2. The point 1 above makes it necessary to have custom format_message for each plugin

File: tendrl/alerting/handlers/cluster/brick_utilization_handler.py (hunk 1 above)
Comment (anmolbabu) [inline tendrl/alerting/handlers/cluster/brick_utilization_handler.py]: Done

File: tendrl/alerting/handlers/cluster/cluster_utilization_handler.py (hunk 2 above)
Comment (anmolbabu) [inline tendrl/alerting/handlers/cluster/cluster_utilization_handler.py]: Done

File: tendrl/alerting/handlers/cluster/osd_utilization_handler.py (hunk 3 above)
Comment (anmolbabu) [inline tendrl/alerting/handlers/cluster/osd_utilization_handler.py]: Done

File: tendrl/alerting/handlers/cluster/pool_utilization_handler.py (hunk 4 above)
Comment (anmolbabu) [inline tendrl/alerting/handlers/cluster/pool_utilization_handler.py]: Done

File: tendrl/alerting/handlers/cluster/volume_utilization_handler.py (hunk 5 above)
Comment (anmolbabu) [inline tendrl/alerting/handlers/cluster/volume_utilization_handler.py]: Done

File: tendrl/alerting/handlers/node/cpu_handler.py (hunk 6 above)
Comment (anmolbabu) [inline tendrl/alerting/handlers/node/cpu_handler.py]: Done

File: tendrl/alerting/handlers/node/memory_handler.py (hunk 7 above)
Comment (anmolbabu) [inline tendrl/alerting/handlers/node/memory_handler.py]: Done

File: tendrl/alerting/handlers/node/mount_point_handler.py (hunk 8 above)
Comment (anmolbabu) [inline tendrl/alerting/handlers/node/mount_point_handler.py]: Done

File: tendrl/alerting/handlers/node/swap_handler.py (hunk 9 above)
Comment (anmolbabu) [inline tendrl/alerting/handlers/node/swap_handler.py]: Done

**Comment by anmolbabu** [comment]
@shtripat Incorporated your comments please revisit the PR
@r0h4n @nthomas-redhat Please review

File: tendrl/alerting/handlers/cluster/brick_utilization_handler.py (hunk 10 above)
Comment (shtripat) [inline tendrl/alerting/handlers/cluster/brick_utilization_handler.py]: ok

File: tendrl/alerting/handlers/cluster/cluster_utilization_handler.py (hunk 2 above)
Comment (r0h4n) [inline tendrl/alerting/handlers/cluster/cluster_utilization_handler.py]: Is this done?

Comment (anmolbabu) [inline tendrl/alerting/handlers/cluster/cluster_utilization_handler.py]: No plugin_instance in use here so no need to check

**Pull request merged**